File: FrontEnd/library-app/src/app/shared/models/Review.ts
import { Book } from "./Book";
import { LibraryUser } from "./LibraryUser";

export interface IReview {
    reviewId: number;
    bookId: number;
    reviewText: string;
    libraryUserId: string;
    rating: number;
    book: Book;
    libraryUser: LibraryUser;
}

export class Review implements IReview {
    reviewId!: number;
    bookId!: number;
    reviewText!: string;
    libraryUserId!: string;
    rating!: number;
    book!: Book;
    libraryUser!: LibraryUser;
}

export class ReviewDTO {
    reviewer!: string;
    reviewText!: string;
    rating!: number;
}

File: BackEnd/LibraryModel/Model/Review.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LibraryModel.Model
{
    public class Review
    {
        public int ReviewId { get; set; }
        public int BookId { get; set; }
        public int Rating { get; set; }
        public string? LibraryUserId { get; set; }
        public string? ReviewText { get; set; }
        public Book? Book { get; set; }
        public LibraryUser? LibraryUser { get; set; }

    }
}
